routes/numbers.js

```diff
@@ -0,0 +1,241 @@
+    (req.header("Content-Type") || req.header("Accept") || "").includes(
+      "application/json"
+    ) === true;
+  function factObjStr() {
+    return JSON.stringify(factObj, null, " ");
+  }
+
+  res.set("X-Numbers-API-Number", factObj.number);
+  res.set("X-Numbers-API-Type", factObj.type);
+  setExpireHeaders(res);
+
+  if (req.query.callback) {
+    // JSONP
+    res.json(useJson ? factObj : factStr);
+  } else if (req.query.write !== undefined) {
+    var arg = useJson ? factObjStr() : '"' + _.escape(factStr) + '"';
+    var script = "document.write(" + arg + ");";
+    res.set("Content-Type", "text/javascript").send(script);
+  } else {
+    if (useJson) {
+      res.set("Content-Type", "application/json").send(factObjStr());
+    } else {
+      res.set("Content-Type", 'text/plain; charset="UTF-8"').send(factStr);
+    }
+  }
+}
+
+// TODO: Refactor to have less duplicated code
+/*
+ * Similar to factResponse(), but supports returning facts for multiple numbers in order
+ * support making batch requests
+ */
+function factsResponse(fact, req, res, nums) {
+  var useJson =
+    req.params.json !== undefined ||
+    (req.header("Content-Type") || "").indexOf("application/json") !== -1;
+  var factsObj = {};
+  _.each(nums, function (num) {
+    const type = req.params.type || "trivia";
+    var factObj = fact.getFact({ number: num, type: type, options: req.query });
+    if (useJson) {
+      factsObj[num] = factObj;
+    } else {
+      factsObj[num] = "" + factObj.text;
+    }
+  });
+
+  function factsObjStr() {
+    return JSON.stringify(factsObj, null, " ");
+  }
+
+  setExpireHeaders(res);
+
+  if (req.query.callback) {
+    // JSONP
+    res.json(factsObj);
+  } else if (req.query.write !== undefined) {
+    var script = "document.write(" + factsObjStr() + ");";
+    res.status(200).set("Content-Type", "text/javascript").send(script);
+  } else {
+    res.status(200).set("Content-Type", "application/json").send(factsObjStr());
+  }
+}
+
+// exports.route = function (app, fact) {
+var allTypesRegex = "/:type(date|year|trivia|math)?";
+
+// parse a batch request string of (e.g. "1..3,10") into individual numbers
+function getBatchNums(rangesStr, parseValue) {
+  var nums = [];
+  var count = 0;
+  var ranges = rangesStr.split(",");
+  _.each(ranges, function (range) {
+    var bounds = range.split("..");
+    if (bounds.length == 1) {
+      if (count == BATCH_LIMIT) {
+        return;
+      }
+      nums.push(parseValue(bounds[0]));
+    } else if (bounds.length == 2) {
+      var minBound = parseValue(bounds[0]);
+      var maxBound = parseValue(bounds[1]);
+      for (var i = minBound; i <= maxBound; i++) {
+        if (count == BATCH_LIMIT) {
+          return;
+        }
+        nums.push(i);
+        count++;
+      }
+    } else {
+      console.log("Unexpected number of bounds in range: " + bounds.length);
+    }
+  });
+  return nums;
+}
+
+// TODO: Precompile this template.
+router.get("/", function (req, res) {
```

Comment:
Since this route renders the main html page and id different than the other api routes, let's keep it in app.js. Also let's have some docstrings for describing what it does.